x-pack/plugins/enterprise_search/server/lib/check_access.ts

```diff
@@ -43,21 +43,18 @@ export const checkAccess = async ({
   request,
   log,
 }: CheckAccess): Promise<ProductAccess> => {
-  const isRbacEnabled = security?.authz.mode.useRbacForRequest(request) ?? false;
+  const isRbacEnabled = security.authz.mode.useRbacForRequest(request);
 
-  // We can only retrieve the active space when either:
-  // 1) security is enabled, and the request has already been authenticated
-  // 2) security is disabled
-  const attemptSpaceRetrieval = !isRbacEnabled || request.auth.isAuthenticated;
+  // If security has been disabled, always hide the plugin
+  if (!isRbacEnabled) {
+    return DENY_ALL_PLUGINS;
+  }
 
-  // If we can't retrieve the current space, then assume the feature is available
+  // We can only retrieve the active space when security is enabled and the request has already been authenticated
+  const attemptSpaceRetrieval = isRbacEnabled && request.auth.isAuthenticated;
```

Comment:
nit: `isRbacEnabled` will always be true if we get to this point, so it's not strictly necessary to check this.

Reply:
++ for removing! I was worried the code comment would look weird if I removed it or that the diff would be confusing, but I think it makes sense to just yeet it outta here

Reply:
https://github.com/elastic/kibana/pull/106307/commits/f65d7721eaf8295f96404f0e617f5ab71f03b0be